pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -77,6 +77,11 @@ func endExecutionSpan(processTrace func(p *api.Payload), endTime time.Time) {
 		Duration: duration,
 	}
 
+	// set the span as an error span if necessary
```

Comment:
I don't think you need this comment -- code is self-explanatory here (that's a good thing)